website/docs/api/plugins/eslint-plugin.md

```diff
@@ -0,0 +1,171 @@
+---
+sidebar_position: 6
+id: eslint-plugin
+title: '📦 eslint-plugin'
+slug: '/api/plugins/@docusaurus/eslint-plugin'
+---
+
+import APITable from '@site/src/components/APITable';
+
+Docusaurus eslint plugin to ensure best Docusaurus practices.
+
+## Installation
+
+```bash npm2yarn
+npm install --save-dev @docusaurus/eslint-plugin
+```
+
+## Usage
+
+Add `@docusaurus` to the plugins section of your `.eslintrc` configuration file:
+
+```json
+{
+  "plugins": ["@docusaurus"]
+}
+```
+
+Then, you can extend one of the configs (e.g. the `recommended` config):
+
+```json
+{
+  "extends": ["plugin:@docusaurus/recommended"]
+}
+```
+
+Each config contains a set of rules. For more fine-grained control, you can also configure the rules you want to use directly:
+
+```json
+{
+  "rules": {
+    "@docusaurus/string-literal-i18n-messages": "error",
+    "@docusaurus/no-untranslated-text": "warn"
+  }
+}
+```
+
+## Supported Configs
+
+### recommended{#config-recommended}
+
+#### Rules:{#config-recommended-rules}
+
+- @docusaurus/string-literal-i18n-messages
+
+### all{#config-all}
+
+#### Rules:{#config-all-rules}
+
+- @docusaurus/string-literal-i18n-messages
+- @docusaurus/no-untranslated-text
+
+## Supported Rules
```

Comment:
Let's split this into individual doc pages to prevent the page from bloating over time, like TS-ESLint's docs: https://typescript-eslint.io/rules/

Reply:
Will do the split, no problem. Just bear in mind that it likely won't reduce the bloat (given that you think https://typescript-eslint.io/rules/ is bloated already).

Reply:
The point is we want to prevent _a single Markdown file_ from bloating. Having a long sidebar isn't a problem because people won't easily lose the scroll position.

Table of contents is more flaky than sidebar. You shouldn't use it as a substitution for sidebar—if a doc page has the potential to grow infinitely long, it's time to split it. Doc sites people hate (at least me, and a few I've talked to): [Joi](https://joi.dev/api/?v=17.6.0), [lodash](https://lodash.com/docs/4.17.15), [TSConfig ref](https://www.typescriptlang.org/tsconfig). Absurdly long pages are extremely hard to navigate and link to, because you have a hard time aiming your cursor at that short scrollbar.

Reply:
> (given that you think https://typescript-eslint.io/rules/ is bloated already).

No, I'm using TS-ESLint as a _positive_ example 😄 A negative example in the ESLint world would be [eslint-plugin-jsdoc](https://github.com/gajus/eslint-plugin-jsdoc)

Reply:
> > (given that you think https://typescript-eslint.io/rules/ is bloated already).
> 
> No, I'm using TS-ESLint as a _positive_ example smile A negative example in the ESLint world would be [eslint-plugin-jsdoc](https://github.com/gajus/eslint-plugin-jsdoc)

Ah ok. "to prevent the page from bloating over time, like TS-ESLint's docs" <-- that's what confused me. Got it now, thanks :)

Reply:
Addressed in 9f04ae1.